### ISSUE TITLE
[regression] Autoawait is broken on 7.30.0

### ISSUE THREAD

**Issue by nzig** [ipython/ipython#13351]
```python
Python 3.9.9 (main, Nov 20 2021, 21:30:06)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.30.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %autoawait
IPython autoawait is `on`, and set to use `asyncio`

In [2]: async def foo(): pass

In [3]: await foo()
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
/usr/lib/python3.9/site-packages/IPython/core/async_helpers.py in __call__(self, coro)
     26         import asyncio
     27
---> 28         return asyncio.get_event_loop().run_until_complete(coro)
     29
     30     def __str__(self):

/usr/lib/python3.9/asyncio/events.py in get_event_loop(self)
    640
    641         if self._local._loop is None:
--> 642             raise RuntimeError('There is no current event loop in thread %r.'
    643                                % threading.current_thread().name)
    644

RuntimeError: There is no current event loop in thread 'MainThread'.
```

This worked on 7.29.0.

**Comment by jsatt** [comment]
I'm seeing identical results with IPython 7.30.0 under Python 3.8. Downgrading to 7.29.0 also fixes it for me.

**Comment by Carreau** [comment]
#13349 should fix it, I'll try to do a 7.31.1
